src/utils/globals.ts

```diff
@@ -11,6 +14,27 @@
 // eslint-disable-next-line no-restricted-globals
 const win: (Window & typeof globalThis) | undefined = typeof window !== 'undefined' ? window : undefined
 
+interface PosthogExtensions {
+    parseErrorAsProperties?: ([event, source, lineno, colno, error]: ErrorEventArgs) => ErrorProperties
+    errorWrappingFunctions?: {
+        wrapOnError: (captureFn: (props: Properties) => void) => () => void
+        wrapUnhandledRejection: (captureFn: (props: Properties) => void) => () => void
+    }
+    rrweb?: { record: any; version: string; rrwebVersion: string }
+    rrwebPlugins?: { getRecordConsolePlugin: any; getRecordNetworkPlugin?: any }
+    canActivateRepeatedly?: (survey: any) => boolean
+    webVitalsCallbacks?: {
+        onLCP: (metric: any) => void
+        onCLS: (metric: any) => void
+        onFCP: (metric: any) => void
+        onINP: (metric: any) => void
+    }
+    tracingHeadersPatchFns?: {
+        _patchFetch: (sessionManager: SessionIdManager) => () => void
+        _patchXHR: (sessionManager: any) => () => void
+    }
+}
```

Comment:
here's the secret sauce... we've added a bunch of stuff directly to window and you can't really see what without reading a bunch of files, and you don't get any typesafety at all since it's all `(window as any)` 

we'd have to be careful here not to bring in the lazy bundles and inflate the bundle unnecessarily

but this is at least a step better